### ISSUE TITLE
 Advanced Tremolo Options (Mainly for percussion)

### ISSUE THREAD

**Issue by zacjansheski** [musescore/MuseScore#20883]
### Your idea

A special window for advanced tremolo options the user can open after right-clicking a tremolo.

The window could contain something like this... Keep in mind I am not a designer :)

_____________________________________________________________________________

☐ Measured    ☐ Unmeasured    ☐ Sustain sound (e.g. closed roll)

☐ Skeleton/rate control:

Notation appearance:
<img width="596" alt="Screenshot 2024-01-08 at 11 06 11 AM" src="https://github.com/musescore/MuseScore/assets/69917893/26365799-c435-43ae-8468-ecd282b92a23">

Playback:
<img width="510" alt="Screenshot 2024-01-08 at 11 06 21 AM" src="https://github.com/musescore/MuseScore/assets/69917893/48bdd181-02b3-4072-a943-40c258c83455">

______________________________________________________________________________________

The user would need to be able to edit the playback image in the same way as writing rhythms and adding articulations/metered tremolo

### Problem to be solved

Across percussion music, there are various philosophies and playback expectations for tremolos/rolls.
A good example would be rudimental vs concert snare drumming, but there are other examples as well.
To accommodate more users writing for percussion, they need to be able to establish their own rules for tremolo notation and playback.

### Additional context

Settings should be able to be copied and pasted throughout the score.  But new tremolos would likely need to be set to default when they are written from scratch.

**Comment by zacjansheski** [comment]
For more context.  The Muse Sounds snare drum plays a sustained buzz roll for the three-slash tremolo.  But it is common to use the two-slash tremolo on eighth notes.  Having that option/control would be desirable.

<img width="1092" alt="Screenshot 2024-01-08 at 12 41 39 PM" src="https://github.com/musescore/MuseScore/assets/69917893/12cd327a-77c0-45d4-a788-95ff96a70d7a">

**Comment by zacjansheski** [comment]
Rudimental/marching music is inconsistent.  It might be desired for this:
<img width="251" alt="Screenshot 2024-01-08 at 12 48 37 PM" src="https://github.com/musescore/MuseScore/assets/69917893/d4263e40-aa72-4a42-a2de-db4fa9672874">


To mean any of these
<img width="652" alt="Screenshot 2024-01-08 at 12 48 42 PM" src="https://github.com/musescore/MuseScore/assets/69917893/0a891b2b-7512-4239-9814-c6dad19c313f">

**Comment by zacjansheski** [comment]
Another example of this (and https://github.com/musescore/MuseScore/issues/20950)

fp crescendo rolls on large drums need an opportunity for the first strike to decay, so the player hits and waits to roll to crescendo.  The example here shows this approach sounds better in the software as well.

https://github.com/musescore/MuseScore/assets/69917893/09151580-d8f7-4195-b09d-972a2030d187

**Comment by bkunda** [comment]
@zacjansheski I think this might have to be something we tackle as the new percussion system evolves over time (in other words, it's probably going to be beyond the scope of the initial feature release, but should be a high priority for subsequent release updates).

**Comment by TobyTwo32** [comment]
@bkunda @zacjansheski this should effect other sections non-percussion, like strings too. The possibilities this opens for better playback without sacrificing the readibility of the score. Currently there are bugs with strings that they don't play tremolos correctly.

**Comment by DaddyLudwig** [comment]
Yeah, with 1 stroke and 2 stroke measured tremolos in strings, they always sound staccatissimo, whereas, at least in slow movements, I'd want the notes of the tremolo to be sustained. The staccatissimo default makes perfect sense in a Presto, but in an Adagio, not so much.

https://github.com/user-attachments/assets/191764d3-0561-4d5b-b351-15e09e173d85

**Comment by TobyTwo32** [comment]
@DaddyLudwig sustain tremolos would sound terrible if they don't have round robins recorded, what would you want in this case is portato samples to play since they hold a bit longer than staccato which would make perfect for repetitions in slow tempos in my honest opinion.

**Comment by wizofaus** [comment]
Measured tremolo is just a notational "convenience" - the playback should be identical to the same notes written out in full.
Unmeasured is different of course, and having a flag or some global threshold to determine whether a tremolo is measured or unmeasured would be handy.
